src/encoder.rs

```diff
@@ -470,7 +470,8 @@ impl FrameInvariants {
         let mut min_partition_size = if config.speed <= 1 { BlockSize::BLOCK_4X4 }
                                  else if config.speed <= 2 { BlockSize::BLOCK_8X8 }
                                  else if config.speed <= 3 { BlockSize::BLOCK_16X16 }
-                                 else { BlockSize::BLOCK_32X32 };
+                                 else if config.speed <= 3 { BlockSize::BLOCK_32X32 }
```

Comment:
This change leaves the final source as
```
else if config.speed <= 3 { BlockSize::BLOCK_16X16 }
else if config.speed <= 3 { BlockSize::BLOCK_32X32 }
```
so mutually exclusive checks for the same condition with different outcomes, one of which is never reached.
Is this intended? Or was the second line meant to be for speed <= 4?

Reply:
It was meant to be <= 4. Nice catch!